### PR TITLE
[RISCV] Add srmcfg CSR from Ssqosid extension.

### PR DESCRIPTION
**Pull request by topperc**
Based on the spec here https://github.com/riscv/riscv-ssqosid/releases/tag/v1.0-rc1

Ssqosid extension name will be added in a separate patch.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVSystemOperands.td [hunk 1]
```diff
@@ -132,6 +132,11 @@ def : SysReg<"sip", 0x144>;
 let DeprecatedName = "sptbr" in
 def : SysReg<"satp", 0x180>;
 
+//===----------------------------------------------------------------------===//
+// Quality-of-Service(QoS) Identifiers (Ssqosid)
+//===----------------------------------------------------------------------===//
+def : SysReg<"srmcfg", 0x181>;
```
Comment (preames) [inline llvm/lib/Target/RISCV/RISCVSystemOperands.td:138]: Two questions.

1) Where in the specification is the register number mentioned?  I didn't find it with some quick searching.

2) Do we need to do anything special to worry about sysregs of different sizes?  This one is SXLEN-bit for instance, and I don't see that obviously being reflected here.

Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVSystemOperands.td:138]: 1. End of the paragraph at the beginning of Chapter 2.  

> The srmcfg register is an SXLEN-bit read/write register used to configure a Resource Control ID (RCID) and a Monitoring Counter ID (MCID). Both RCID and MCID are WARL fields. The register is
formatted as shown in Figure 1 when SXLEN=64 and Figure 2 when SXLEN=32. **The CSR number is
0x181**.

2. I don't think so. The only thing I know about is registers that have a high register for RV32.

Comment (preames) [reply llvm/lib/Target/RISCV/RISCVSystemOperands.td:138]: On the register number, I swear I grepped for those words.  Thanks.

**Review by preames — APPROVED** [review]
LGTM

**Pull request merged**